.github/workflows/code-format-check.yml

```diff
@@ -0,0 +1,31 @@
+name: Code format check
+
+on: push
+
+jobs:
+  test:
+
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.11
+
+      - name: Install Poetry
+        run: |
+          curl -sSL https://install.python-poetry.org | python3 -
+
+      - name: Install project
+        run: |
+          # We only need Black, and we want to use the project-specific version.
+          # (Defaults in Black change from time to time and we want only to deal with
+          # the consequences, if any, on our own schedule.)
+          poetry install --only dev
+
+      - name: Run check
+        run: poetry run black . --check
```

Comment:
I see the utility of separating this check out. It's unfortunate that it requires *so* much boilerplate for one extra command.

Reply:
A partial solution might be to create a workflow action that
- checks out the repo
- sets up Python
- installs Poetry

Reply:
But of course that comes with its own burden of dev, testing, maintenance, etc. Small but not negligible.

Reply:
Yeah... the fact that we install poetry in every action in every repo has been bugging me. I have it on my TODO list to look at at a higher level.